### PR TITLE
Gd dsdeqa 179

### PR DESCRIPTION
**Pull request by gdlex4015**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by andy7i** [comment]
Is this to solve the problem where each test cases within a test suite will run sequentially?
A cleaner way to parallelize might be to use this trait that will run test cases within a suite in parallel: 
http://doc.scalatest.org/3.0.1/index.html#org.scalatest.ParallelTestExecution

**Comment by aweng98** [comment]
That's not the issue. We're working to refactor big test suite into multiple smaller sized test suites. The goal is have more test suites run in parallel. trait `ParallelTestExecution` cannot be used because existing trait `cleanup` doesn't support parallel tests execution.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceReaderSpec.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                workspacePage.hasGoogleBillingLink shouldBe false
+                workspacePage.hasStorageCostEstimate shouldBe false
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceReaderSpec.scala]: need a newline to mark end of file

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceWriterSpec.scala [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                  }
+                }
+              }(authTokenOwner)
+            }
+          }(authTokenOwner)
+        }
+      }
+    }
+  }
+}
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceWriterSpec.scala]: new line

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceReaderSpec.scala [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            workspacesTable.readColumnHeaders shouldBe List("Status", "Workspace", "Description", "Last Modified", "Access Level")
+          }
+        }
+      }
+    }
+  }
+
+  "A user" - {
+    "who has reader access to workspace" - {
+      "should see launch analysis button disabled" in {
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceReaderSpec.scala:59]: this test and `"should see import config button disabled"` have same workspace signature. if you add the verification step from `"should see import config button disabled"` to this test, then you combined two into one.  I'm fine without combine. you make the call.

**Review by aweng98 — APPROVED** [review]

**Comment by andy7i** [comment]
@aweng98 
> The goal is have more test suites run in parallel

I think we're agreeing here. The end goal is to have more test cases running in parallel right? Right now we're trying to solve that by separating test cases out into more suites since it currently runs tests in a suite sequentially. 

> trait cleanup doesn't support parallel tests execution

Where is this? As I understand it, ParallelTestExecution should run each test case as its own instance and handle before and after for each test case.

**Comment by aweng98** [comment]
`with CleanUp` is `object` class. single instance for all caller.

I experimented `with ParallelTestExecution` before with a different test suite. Ended up with random and unexplainable test failures. But I'm not 100% certain whether same outcome for `WorkspaceSpec`. I assumed it's the same outcome based on my understanding.

**Review by andy7i — APPROVED** [review]
👍 assuming tests pass

**Comment by gdlex4015** [comment]
jenkins retest

**Comment by gdlex4015** [comment]
jenkins retest

**Pull request merged**